addons/product/models/product_attribute.py

```diff
@@ -78,6 +78,12 @@ def _compute_products(self):
     def _without_no_variant_attributes(self):
         return self.filtered(lambda pa: pa.create_variant != 'no_variant')
 
+    @api.onchange("display_type", "number_related_products")
+    def _onchange_display_type(self):
```

Comment:
Why you use on-change can't we use `compute` method? :thinking: 
Because on-change will only trigger from form view and compute can be trigger from anywhere!
Or do we want this change only when we change something from form not from other places :sleepy: